lib/imap.ml

```diff
@@ -706,6 +706,14 @@ module D = struct
     if is_atom_char $ cur d then p_atomf "NIL" $ k [] $ d else
     p_ch '(' (fun d -> if cur d = ')' then readc $ k [] $ d else p (fun x -> loop [x]) d) d
 
+  let p_list_nosp p k d =
```

Comment:
Is it possible to factor out the common code of `p_list` and `p_list_nosp` by letting `p_list` take a parameter specifying a string to use as a separator ?